paywall/src/unlockTypes.ts

```diff
@@ -0,0 +1,52 @@
+// This file contains type definitions for the various kinds of data that we use
+// throughout unlock-app.
+
+// A bug in eslint causes it to think that this exported enum is "unused". So
+// disable eslint for that declaration until they fix it. TODO: follow up on this.
+/* eslint-disable no-unused-vars */
+export enum TransactionType {
+  LOCK_CREATION = 'LOCK_CREATION',
+  KEY_PURCHASE = 'KEY_PURCHASE',
+  WITHDRAWAL = 'WITHDRAWAL',
+  UPDATE_KEY_PRICE = 'UPDATE_KEY_PRICE',
+}
+/* eslint-enable no-unused-vars */
+
+/* eslint-disable no-unused-vars */
+export enum TransactionStatus {
+  SUBMITTED = 'submitted',
+  PENDING = 'pending',
+  MINED = 'mined',
+}
+/* eslint-enable no-unused-vars */
+
+export interface Transaction {
+  status: TransactionStatus
+  confirmations: number
+  hash: string
+  type: TransactionType
+  blockNumber: number
+
+  lock?: string
+  name?: string
+  key?: string // TODO: tighten up our types, hopefully we won't have too many
+  // optional properties.
+}
+
+// Analogous to the transactions in Redux
+export interface Transactions {
+  [hash: string]: Transaction
```

Comment:
How strongly do we believe that this type declaration appropriately represents a collection of Transactions as they exist today?

Reply:
For the collection part, I think 100%. They're kept as a mapping from hashes to transactions. For the actual individual transactions, somewhat less confident. This definition basically started with what was always in Redux, and grew some optional properties because sometimes transactions get extra ones. There may be others that were missed.